### PR TITLE
Cython implementation of kuiper_two.

### LINKED ISSUE #8098: Fast implementation of kuiper_two
The kuiper_two implementation currently contains comments about efficiency:
```
    # this could be more efficient  # <--- this
    cdfv1 = np.searchsorted(data2, data1) / float(len(data2))
    # this could be more efficient
    cdfv2 = np.searchsorted(data1, data2) / float(len(data1))
```

In case you are interested, I PR'd a Cython implementation of scipy.stats.ks_2samp to scipy (https://github.com/scipy/scipy/pull/9462), which is around 35-45% faster for N~1e4-1e5; in fact a nearly identical PR did go into scipy a while ago (https://github.com/scipy/scipy/pull/5938) but was reverted because it did not handle ties in the data "in the same way as the previous implementation" (even though ties are not really consistent with the underlying statistical theory, which assumes continuous distributions -- see discussions in the scipy tracker); the new implementation handles ties "correctly".

Obviously that implementation could be nearly directly adapted to compute the Kuiper 2-sample statistic.

Not making a PR for astropy as I'm a bit too lazy right now to figure out how astropy handles their cython parts, but if anyone is interested feel free to steal that implementation.

### PR DESCRIPTION
**Pull request by anntzer**
Gives a \~20-30% speedup at N~1e4-1e5.

Test script:

    from time import perf_counter
    import numpy as np
    from astropy.stats import kuiper_two

    N = 100000
    np.random.seed(0)

    start = perf_counter()
    for _ in range(100):
        x, y = np.random.random((2, N))
    generation_time = perf_counter() - start

    checksum = 0
    start = perf_counter()
    for _ in range(100):
        x, y = np.random.random((2, N))
        d, fpp = kuiper_two(x, y)
        checksum += d
    print(perf_counter() - start - generation_time)
    print(checksum)

Closes #8098.

### CODE REVIEW

**Comment by crawfordsm** [comment]
Thanks @anntzer !    I've restarted the test as the failure doesn't look related to your contribution.   

This could use an entry in the change log, but otherwise, this looks great!

**Comment by anntzer** [comment]
Should it go straight to docs/whatsnew/3.2.rst?

**Comment by pllim** [comment]
@anntzer , please add the entry in `CHANGES.rst`. Given that this is neither new feature, nor API change, nor bug fix, maybe put it in "other additions"?

**Comment by anntzer** [comment]
done

File: CHANGES.rst [hunk 1]
```diff
@@ -213,9 +213,12 @@ astropy.wcs
...
 ---------------------------
 
-- Nothing changed yet.
-
+astropy.stats
+^^^^^^^^^^^^^
 
+- A Cython implementation for `astropy.stats.kuiper_two` and a vectorized
+  implementation for `astropy.stats.kuiper_false_positive_probability` have
+  been added, speeding up both functions.
```
Comment (pllim) [inline CHANGES.rst]: Please append with a "[#8104]" at the end to satisfy bot. Thanks for your patience! (If in doubt, see other entries.)

Comment (anntzer) [reply CHANGES.rst]: done

**Pull request merged**